python/cxx/deck_keyword.cpp

```diff
@@ -56,11 +59,80 @@ struct DeckRecordIterator
     }
 };
 
+
+bool is_number(const std::string& s)
+{
+    return !s.empty() && std::find_if(s.begin(), 
+        s.end(), [](char c) { return !std::isdigit(c); }) == s.end();
+}
+
+
+void push_string_as_deck_value(std::vector<DeckValue>& record, const std::string str) {
+
+    int max_indx = str.length() - 1;
+    if (str[max_indx] == '*') {
+        int it;
+
+        if (max_indx == 0)
+            it = 1;
+        else {
+            std::string substr = str.substr(0, max_indx);
+            if (is_number(substr)) {
+                it = std::stoi(substr);
+            }
+            else
+                throw py::type_error("DeckKeyword: string values ending in '*' can only be preceeded by a number.");
+        }
+
+        for (int i = 0; i < it; i++)
+            record.push_back( DeckValue() );
+    }
+    else
+        record.push_back( DeckValue(str) );
+}
+
+
 }
 
 void python::common::export_DeckKeyword(py::module& module) {
     py::class_< DeckKeyword >( module, "DeckKeyword")
         .def(py::init<const ParserKeyword& >())
+
+        .def(py::init([](const ParserKeyword& parser_keyword, py::list record_list) {
+
+            std::vector< std::vector<DeckValue> > value_record_list;
+
+            for (py::handle record_obj : record_list) {
+                 py::list record = record_obj.cast<py::list>();
+                 std::vector<DeckValue> value_record;
+
+                 for (py::handle value_obj : record) { 
+
+                      try {
```

Comment:
Instead of nesting the `try catch`blocks I think this would be easier to read if you used the `continue`statement:
```
try {
     int val_int = value_obj.cast<int>();
     value_record.emplace_back( val_int );
     continue;
} catch (std::exception) { }

try { 
     double val_double = value_obj.cast<double>();
     ...
     continue;
}
...
```